### PR TITLE
Fix update generation with additionalIdentifiers

### PR DESCRIPTION
**Pull request by ptrucks**
If the `additionalIdentifiers` property is present in a resource schema
the `generate_update_example` method in `resource_client` will not work,
because it checks for unique keys from the create_model and passes
`self._additional_identifiers_paths` to a method that expects a list of
tuples of string, whereas it receives a list of sets of tuples of
string and will fail with

```
AttributeError: 'tuple' object has no attribute 'split'
```

Instead, check if the key is in any of the additional identifiers.

Related issue
-------------

https://github.com/aws-cloudformation/cloudformation-cli/issues/731

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ptrucks** [comment]
Please note that contract tests are failing because the workflow can't access the secrets to retrieve credentials, not because the tests themselves are failing.

File: tests/contract/test_resource_client.py [hunk 1]
```diff
@@ -80,6 +80,43 @@
     "primaryIdentifier": ["/properties/c", "/properties/d"],
 }
 
+SCHEMA_WITH_TRANSFORM = {
```
Comment (prerna-p) [inline tests/contract/test_resource_client.py:83]: Why is this needed?

Comment (ptrucks) [reply tests/contract/test_resource_client.py:83]: This is not part of my change, but got pulled into the diff due to a merge when rebasing.

Comment (ptrucks) [reply tests/contract/test_resource_client.py]: Removed the extraneous lines from the commit, sorry for the noise.

**Review by prerna-p — APPROVED** [review]

File: tests/contract/test_resource_client.py [hunk 2]
```diff
@@ -202,6 +215,51 @@ def resource_client_inputs_schema():
...
+                EMPTY_OVERRIDE,
+                {
+                    "CREATE": {"a": 111, "c": 2, "d": 3},
+                    "UPDATE": {"a": 1, "c": 2},
+                    "INVALID": {"c": 3},
+                },
+            )
+
+    mock_sesh.client.assert_called_once_with("lambda", endpoint_url=endpoint)
+    mock_creds.assert_called_once_with(mock_sesh, LOWER_CAMEL_CRED_KEYS, None)
```
Comment (anshikg) [inline tests/contract/test_resource_client.py:252]: Can we reuse an existing resource client vs creating a new one?

Comment (ptrucks) [reply tests/contract/test_resource_client.py:252]: Sure, I was following the style of the other tests at the time but it seems like that has been changed since. I'll update to the new style.

Comment (ptrucks) [reply tests/contract/test_resource_client.py]: I have addressed this by parameterizing the fixture, let me know if that works for you.

**Review by anshikg — APPROVED** [review]
